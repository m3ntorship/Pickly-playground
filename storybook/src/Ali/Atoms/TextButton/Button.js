import React from "react";
import PropTypes from "prop-types";

const Button = (props) => {
  const {
    textButtonDisabled,
    textButtonsize,
    textButtonvariant,
    textButtonchildren,
  } = props;

  return (
    <button
      disabled={textButtonDisabled}
      className={`capitalize rounded-full focus:outline-none disabled:opacity-25 ${
        textButtonvariant === "Primary"
          ? `bg-primary-shd1  text-white ${
              !textButtonDisabled && "hover:bg-primary"
            } focus:border-primary focus:border focus:ring-1 focus:ring-primary-shd1`
          : `border text-primary-shd1 border-primary-shd1 ${
              !textButtonDisabled && "hover:bg-grey-bg"
            } focus:ring-1 focus:ring-primary-shd1`
      } ${
        textButtonsize === "Large"
          ? "text-md font-bold px-xl py-3.5 "
          : textButtonsize === "Medium"
          ? "text-base font-medium px-l py-xs "
          : "text-sm font-medium px-m py-1.5 "
      }
      ${textButtonDisabled && "cursor-not-allowed"}`}
    >
      {textButtonchildren}
    </button>
  );
};

Button.propTypes = {
  textButtonsize: PropTypes.oneOf(["Small", "Medium", "Large"]),
  textButtonvariant: PropTypes.oneOf(["Primary", "Secondary"]),
  textButtonDisabled: PropTypes.bool,
  textButtonchildren: PropTypes.string,
};
export default Button;
