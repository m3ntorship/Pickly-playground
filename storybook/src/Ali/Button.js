import React from "react";
import PropTypes from "prop-types";

const Button = (props) => {
  const { disabled, size, variant, children } = props;
  return (
    <button
      disabled={disabled}
      className={`capitalize rounded-full font-bold focus:outline-none disabled:opacity-25 ${
        variant === "Primary"
          ? `bg-primary-shd1  text-white ${
              !disabled && "hover:bg-primary"
            } focus:border-primary focus:border focus:ring-1 focus:ring-primary-shd1`
          : `border text-primary-shd1 border-primary-shd1 ${
              !disabled && "hover:bg-grey-bg"
            } focus:ring-1 focus:ring-primary-shd1`
      } ${
        size === "Large"
          ? "text-md px-xl py-s "
          : size === "Medium"
          ? "text-base px-l py-xs "
          : "text-sm px-m py-xxs "
      }
      ${disabled ? "cursor-not-allowed" : ""}`}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  size: PropTypes.oneOf(["Small", "Medium", "Large"]),
  variant: PropTypes.oneOf(["Primary", "Secondary"]),
  disabled: PropTypes.bool,
  children: PropTypes.string,
};
export default Button;
