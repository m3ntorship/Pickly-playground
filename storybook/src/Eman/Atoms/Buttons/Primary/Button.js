import React from "react";
import PropTypes from "prop-types";

export const BUTTON_OPTIONS = {
  SIZE: {
    SMALL: "small",
    MEDIUM: "medium",
    BIG: "big",
  },
  TYPE: {
    PRIMARY: "primary",
    SECONDARY: "secondary",
  },
  DISABLED: {
    ON: "on",
    OFF: "off",
  },
};

export const Button = ({ type, content, size, disabled }) => {
  return (
    <button
      className={`
  ${
    type === BUTTON_OPTIONS.TYPE.PRIMARY
      ? "rounded-full  bg-primary  text-white hover:bg-primary-shd8 focus:outline-none focus:border-2 focus:border-primary-shd9 focus:border-opacity-100 disabled:opacity-5"
      : "rounded-full  text-primary  border-2 border-primary"
  },
        ${size === BUTTON_OPTIONS.SIZE.BIG && "px-xl py-s font-bold text-md"},
        ${
          size === BUTTON_OPTIONS.SIZE.MEDIUM &&
          "px-l py-xs font-semibold  text-base"
        },
        ${
          size === BUTTON_OPTIONS.SIZE.SMALL &&
          "px-m py-xxs font-semibold text-sm"
        },
        ${disabled === BUTTON_OPTIONS.DISABLED.ON && "opacity-25"}
        `}
    >
      {content}
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.string,
  content: PropTypes.string,
  size: PropTypes.string,
  disabled: PropTypes.string,
};
