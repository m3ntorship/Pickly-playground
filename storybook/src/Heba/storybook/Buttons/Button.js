import React from "react";
import cn from "classnames";

export const BUTTON_OPTIONS = {
  SIZE: {
    SMALL: "small",
    MEDIUM: "medium",
    BIG: "big",
  },
};

export const Button = ({ primary, size }) => {
  return (
    <div className="flex justify-start items-center ">
      <button
        type="button"
        className={cn(
          "flex  items-center justify-center",
          primary
            ? "text-white bg-primary rounded-full focus:ring-2 focus:ring-primary-shd9 focus:outline-none disabled:opacity-25 mr -4"
            : " text-primary bg-white  border border-primary rounded-full  focus:ring-2 focus:ring-primary-shd9 focus:outline-none mr-4 disabled:opacity-25 fill-current",
          { "py-3 px-3 ": size === BUTTON_OPTIONS.SIZE.BIG },
          {
            "py-2 px-2": size === BUTTON_OPTIONS.SIZE.MEDIUM,
          },

          {
            "py-2 px-2": size === BUTTON_OPTIONS.SIZE.SMALL,
          }
        )}
        disabled
      >
        <svg
          className={cn(
            { "w-8 h-auto ": size === BUTTON_OPTIONS.SIZE.BIG },
            {
              "w-auto h-6": size === BUTTON_OPTIONS.SIZE.MEDIUM,
            },

            {
              "w-3 h-auto": size === BUTTON_OPTIONS.SIZE.SMALL,
            }
          )}
          viewBox="0 0 32 32"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.724 12.3906L16 18.1146L10.276 12.3906L8.39062 14.276L16 21.8853L23.6093 14.276L21.724 12.3906Z"
            className={cn(
              primary ? "text-white fill-current" : "text-primary fill-current"
            )}
          />
        </svg>
      </button>
    </div>
  );
};
