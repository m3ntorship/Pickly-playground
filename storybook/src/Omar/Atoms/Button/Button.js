import React from "react";
import PropTypes from "prop-types";

export const Button = (props) => {
  const {
    disabled = false,
    size = "sm",
    variant = "primary",
    content,
    onClick,
  } = props;

  return (
    <button
      data-testid="btn-case"
      disabled={disabled}
      onClick={onClick}
      className={`rounded-full border disabled:opacity-30 flex items-center ${
        variant === "primary"
          ? "bg-primary text-white hover:bg-primary-shd9 focus:outline-none focus:border-primary-shd9"
          : "bg-white text-primary border-primary border-solid hover:bg-grey-bg2 focus:border-2 focus:outline-none"
      } ${
        size === "lg"
          ? "text-md py-3.5 px-xl"
          : size === "sm"
          ? "text-sm py-1.5 px-m"
          : "text-base py-xs px-l"
      }`}
    >
      <span>{content}</span>
    </button>
  );
};

Button.propTypes = {
  size: PropTypes.oneOf(["sm", "md", "lg"]),
  disabled: PropTypes.bool,
  variant: PropTypes.oneOf(["primary", "secondary"]),
  className: PropTypes.string,
  content: PropTypes.string,
};
