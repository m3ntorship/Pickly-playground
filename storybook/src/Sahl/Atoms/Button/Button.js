import React from "react";
import PropTypes from "prop-types";
import DownArrow from "../../assets/icons/DownArrow";

function Button(props) {
  const {
    disabled = false,
    size = "md",
    variant = "primary",
    leftArrow = false,
    rightArrow = false,
    className,
    children,
  } = props;

  return (
    <button
      disabled={disabled}
      className={`${className} rounded-full border disabled:opacity-30 flex items-center ${
        variant === "primary"
          ? "bg-primary text-white hover:bg-primary-shd9 focus:border-primary-shd9 focus:border-solid focus:outline-none"
          : "bg-white text-primary border-primary border-solid hover:bg-grey-bg2 focus:border-2 focus:outline-none"
      } ${
        size === "lg"
          ? "py-3.5 px-8"
          : size === "sm"
          ? "py-1.5 px-4"
          : "py-2.5 px-6"
      }`}
    >
      {leftArrow && (
        <DownArrow
          path1Fill={variant === "primary" ? "#FFF" : "rgba(112, 72, 232)"}
        />
      )}
      <span>{children}</span>
      {rightArrow && (
        <DownArrow
          path1Fill={variant === "primary" ? "#FFF" : "rgba(112, 72, 232)"}
        />
      )}
    </button>
  );
}

Button.propTypes = {
  size: PropTypes.oneOf(["sm", "md", "lg"]),
  disabled: PropTypes.bool,
  leftArrow: PropTypes.bool,
  rightArrow: PropTypes.bool,
  variant: PropTypes.oneOf(["primary", "secondary"]),
  className: PropTypes.string,
  children: PropTypes.string,
};

export default Button;
