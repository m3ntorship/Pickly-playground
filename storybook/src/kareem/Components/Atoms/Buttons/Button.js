import React from "react";
import PropTypes from "prop-types";
import clsname from "classnames";

export const BUTTON_OPTIONS = {
  SIZE: {
    SMALL: "small",
    MEDIUM: "medium",
    LARGE: "large",
  },
};

export function Button({ primary, children, size }) {
  return (
    <div>
      <button
        type="button"
        className={clsname(
          primary
            ? "bg-primary text-white font-sans rounded-full focus:ring-2 focus:ring-primary-shd9 focus:outline-none disabled:opacity-25 mr -4"
            : "text-primary bg-white font-sans border border-primary rounded-full  focus:ring-2 focus:ring-primary-shd9 focus:outline-none mr-4 disabled:opacity-25",
          {
            "py-3.5 px-8 font-bold text-md": size === BUTTON_OPTIONS.SIZE.LARGE,
          },
          {
            "py-2 px-6 text-base font-medium":
              size === BUTTON_OPTIONS.SIZE.MEDIUM,
          },
          {
            "py-1.5 px-4 font-medium text-sm":
              size === BUTTON_OPTIONS.SIZE.SMALL,
          }
        )}
      >
        {children}
      </button>
    </div>
  );
}

Button.propTypes = {
  /**
   * Is this the principal call to action on the page?
   */
  primary: PropTypes.bool,
  /**
   * What background color to use
   */
  children: PropTypes.string,
  /**
   * How large should the button be?
   */
  size: PropTypes.oneOf(["small", "medium", "large"]),
};

Button.defaultProps = {
  primary: true,
  size: "medium",
  children: this.props.children,
};

export default Button;
