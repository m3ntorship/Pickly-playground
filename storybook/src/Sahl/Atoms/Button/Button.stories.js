import React from "react";

import Button from "./Button";

const defaultExport = {
  title: "Sahl/Atoms/Button",
  component: Button,
};

export default defaultExport;

const Template = (args) => <Button {...args} />;

export const Default = Template.bind({});
Default.args = {
  size: "small",
  variant: "primary",
  disabled: false,
  children: "Button Text",
};
