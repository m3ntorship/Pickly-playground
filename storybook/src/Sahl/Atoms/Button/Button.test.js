import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import Button from "./Button";

describe("Button Component", () => {
  it("should render without any props", () => {
    const tree = renderer.create(<Button>Test Button</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render disabled when passing disabled", () => {
    const tree = renderer
      .create(<Button disabled>Test Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with children when passing it", () => {
    const tree = renderer.create(<Button>Test Button</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with left Arrow when passing leftArrow", () => {
    const tree = renderer
      .create(<Button leftArrow>Test Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with right arrow when passing rightArrow", () => {
    const tree = renderer
      .create(<Button rightArrow>Test Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with both left & right arrow when passing leftArrow & rightArrow", () => {
    const tree = renderer
      .create(
        <Button leftArrow rightArrow>
          Test Button
        </Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with medium size as default when no size is passed", () => {
    const tree = renderer.create(<Button>Test Button</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with small size when size="sm" is passed', () => {
    const tree = renderer
      .create(<Button size="sm">Test Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with large size when size="lg" is passed', () => {
    const tree = renderer
      .create(<Button size="lg">Test Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with primary variant as default without passing any props", () => {
    const tree = renderer.create(<Button>Test Button</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with secondary variant when variant="secondary"', () => {
    const tree = renderer
      .create(<Button variant="secondary">Test Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Button Component DOM testing", () => {
  it("renders Button component", () => {
    // screen.debug();
    const { rerender } = render(<Button>Click here</Button>);
    // Implicit as it returns false if it is not there
    // screen.getByText("Click here");
    // Explicit and recommended
    expect(screen.getByText("Click here")).toBeInTheDocument();
    // re-render the same component with different props
    rerender(<Button>Register</Button>);
    expect(screen.getByText("Register")).toBeInTheDocument();
  });
});
