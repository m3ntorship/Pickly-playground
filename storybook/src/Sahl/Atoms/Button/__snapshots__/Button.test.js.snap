// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render disabled when passing disabled 1`] = `
<button
  className="undefined rounded-full border disabled:opacity-30 flex items-center bg-primary text-white hover:bg-primary-shd9 focus:border-primary-shd9 focus:border-solid focus:outline-none py-2.5 px-6"
  disabled={true}
>
  <span>
    Test Button
  </span>
</button>
`;

exports[`Button Component should render with both left & right arrow when passing leftArrow & rightArrow 1`] = `
<button
  className="undefined rounded-full border disabled:opacity-30 flex items-center bg-primary text-white hover:bg-primary-shd9 focus:border-primary-shd9 focus:border-solid focus:outline-none py-2.5 px-6"
  disabled={false}
>
  <svg
    className="w-6 h-6 undefined"
    fill="none"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16.293 9.29297L12 13.586L7.70697 9.29297L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.29297Z"
      fill="#5B6366"
    />
  </svg>
  <span>
    Test Button
  </span>
  <svg
    className="w-6 h-6 undefined"
    fill="none"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16.293 9.29297L12 13.586L7.70697 9.29297L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.29297Z"
      fill="#5B6366"
    />
  </svg>
</button>
`;

exports[`Button Component should render with children when passing it 1`] = `
<button
  className="undefined rounded-full border disabled:opacity-30 flex items-center bg-primary text-white hover:bg-primary-shd9 focus:border-primary-shd9 focus:border-solid focus:outline-none py-2.5 px-6"
  disabled={false}
>
  <span>
    Test Button
  </span>
</button>
`;

exports[`Button Component should render with large size when size="lg" is passed 1`] = `
<button
  className="undefined rounded-full border disabled:opacity-30 flex items-center bg-primary text-white hover:bg-primary-shd9 focus:border-primary-shd9 focus:border-solid focus:outline-none py-3.5 px-8"
  disabled={false}
>
  <span>
    Test Button
  </span>
</button>
`;

exports[`Button Component should render with left Arrow when passing leftArrow 1`] = `
<button
  className="undefined rounded-full border disabled:opacity-30 flex items-center bg-primary text-white hover:bg-primary-shd9 focus:border-primary-shd9 focus:border-solid focus:outline-none py-2.5 px-6"
  disabled={false}
>
  <svg
    className="w-6 h-6 undefined"
    fill="none"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16.293 9.29297L12 13.586L7.70697 9.29297L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.29297Z"
      fill="#5B6366"
    />
  </svg>
  <span>
    Test Button
  </span>
</button>
`;

exports[`Button Component should render with medium size as default when no size is passed 1`] = `
<button
  className="undefined rounded-full border disabled:opacity-30 flex items-center bg-primary text-white hover:bg-primary-shd9 focus:border-primary-shd9 focus:border-solid focus:outline-none py-2.5 px-6"
  disabled={false}
>
  <span>
    Test Button
  </span>
</button>
`;

exports[`Button Component should render with primary variant as default without passing any props 1`] = `
<button
  className="undefined rounded-full border disabled:opacity-30 flex items-center bg-primary text-white hover:bg-primary-shd9 focus:border-primary-shd9 focus:border-solid focus:outline-none py-2.5 px-6"
  disabled={false}
>
  <span>
    Test Button
  </span>
</button>
`;

exports[`Button Component should render with right arrow when passing rightArrow 1`] = `
<button
  className="undefined rounded-full border disabled:opacity-30 flex items-center bg-primary text-white hover:bg-primary-shd9 focus:border-primary-shd9 focus:border-solid focus:outline-none py-2.5 px-6"
  disabled={false}
>
  <span>
    Test Button
  </span>
  <svg
    className="w-6 h-6 undefined"
    fill="none"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16.293 9.29297L12 13.586L7.70697 9.29297L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.29297Z"
      fill="#5B6366"
    />
  </svg>
</button>
`;

exports[`Button Component should render with secondary variant when variant="secondary" 1`] = `
<button
  className="undefined rounded-full border disabled:opacity-30 flex items-center bg-white text-primary border-primary border-solid hover:bg-grey-bg2 focus:border-2 focus:outline-none py-2.5 px-6"
  disabled={false}
>
  <span>
    Test Button
  </span>
</button>
`;

exports[`Button Component should render with small size when size="sm" is passed 1`] = `
<button
  className="undefined rounded-full border disabled:opacity-30 flex items-center bg-primary text-white hover:bg-primary-shd9 focus:border-primary-shd9 focus:border-solid focus:outline-none py-1.5 px-4"
  disabled={false}
>
  <span>
    Test Button
  </span>
</button>
`;

exports[`Button Component should render without any props 1`] = `
<button
  className="undefined rounded-full border disabled:opacity-30 flex items-center bg-primary text-white hover:bg-primary-shd9 focus:border-primary-shd9 focus:border-solid focus:outline-none py-2.5 px-6"
  disabled={false}
>
  <span>
    Test Button
  </span>
</button>
`;
